violates JLS formatting for type annotations

prettier violates the JLS which states that a type annotation should be just before the type

**Prettier-Java 0.5.1**

**Input:**

```java
  private @Nullable Neo4j neo4j;
```

**Output:**

```java
  @Nullable
  private Neo4j neo4j;
```

**Expected behavior:**

```java
  private @Nullable Neo4j neo4j;
```

![Screenshot from 2022-01-10 13-00-02](https://user-images.githubusercontent.com/5517/148818721-c8b8ef3c-dea8-45dd-a3fa-8c8cf981bf0b.png)

ironic you should mention error prone, because that's what's checking this error

```
/home/xeno/IdeaProjects/pm/backend/module/util/model-graph/src/testFixtures/java/com/xenoterracide/ppm/util/modelgraph/Neo4jExtension.java:18: error: [AnnotationPosition] @Nullable is a TYPE_USE annotation, so should appear after modifiers and directly before the type.
  private Neo4j neo4j;
                ^
    (see https://google.github.io/styleguide/javaguide.html#s4.8.5-annotations)
  Did you mean to remove this line?
1 error
```

--- a/packages/prettier-plugin-java/src/printers/printer-utils.ts
+++ b/packages/prettier-plugin-java/src/printers/printer-utils.ts
@@ -225,18 +225,19 @@ export function sortModifiers(modifiers: CstNode[] | undefined) {
 
   /**
    * iterate in reverse order because we special-case
-   * method annotations which come after all other
+   * type annotations which come after all other
    * modifiers
    */
   forEachRight(modifiers, modifier => {
     const isAnnotation = modifier.children.annotation !== undefined;
-    const isMethodAnnotation =
+    const isTypeAnnotation =
       isAnnotation &&
       (modifier.name === "methodModifier" ||
-        modifier.name === "interfaceMethodModifier");
+        modifier.name === "interfaceMethodModifier" ||
+        modifier.name === "fieldModifier");
 
     if (isAnnotation) {
-      if (isMethodAnnotation && !hasOtherModifier) {
+      if (isTypeAnnotation && !hasOtherModifier) {
         lastAnnotations.unshift(modifier);
       } else {
         firstAnnotations.unshift(modifier);
@@ -424,7 +425,8 @@ function needLineClassBodyDeclaration(
       classMemberDeclaration.children.fieldDeclaration[0];
     if (
       fieldDeclaration.children.fieldModifier !== undefined &&
-      hasAnnotation(fieldDeclaration.children.fieldModifier)
+      hasAnnotation(fieldDeclaration.children.fieldModifier) &&
+      hasNonTrailingAnnotation(fieldDeclaration.children.fieldModifier)
     ) {
       return true;
     }
@@ -443,7 +445,8 @@ function needLineInterfaceMemberDeclaration(
     const constantDeclaration = declaration.children.constantDeclaration[0];
     if (
       constantDeclaration.children.constantModifier !== undefined &&
-      hasAnnotation(constantDeclaration.children.constantModifier)
+      hasAnnotation(constantDeclaration.children.constantModifier) &&
+      hasNonTrailingAnnotation(constantDeclaration.children.constantModifier)
     ) {
       return true;
     }
@@ -487,27 +490,37 @@ function isInterfaceMemberASemicolon(
 }
 
 function hasAnnotation(
-  modifiers: FieldModifierCstNode[] | ConstantModifierCstNode[]
+  modifiers: (
+    | MethodModifierCstNode
+    | InterfaceMethodModifierCstNode
+    | FieldModifierCstNode
+    | ConstantModifierCstNode
+  )[]
 ) {
   return modifiers.some(modifier => modifier.children.annotation !== undefined);
 }
 
 /**
- * Return true if there is a method modifier that does not come after all other modifiers
+ * Return true if there is a modifier that does not come after all other modifiers
  * It is useful to know if sortModifiers will add an annotation before other modifiers
  *
- * @param methodModifiers
+ * @param modifiers
  * @returns {boolean}
  */
 function hasNonTrailingAnnotation(
-  methodModifiers: (MethodModifierCstNode | InterfaceMethodModifierCstNode)[]
+  modifiers: (
+    | MethodModifierCstNode
+    | InterfaceMethodModifierCstNode
+    | FieldModifierCstNode
+    | ConstantModifierCstNode
+  )[]
 ) {
   const firstAnnotationIndex = findIndex(
-    methodModifiers,
+    modifiers,
     modifier => modifier.children.annotation !== undefined
   );
   const lastNonAnnotationIndex = findLastIndex(
-    methodModifiers,
+    modifiers,
     modifier => modifier.children.annotation === undefined
   );
 

--- a/packages/prettier-plugin-java/test/unit-test/blank_lines/_input.java
+++ b/packages/prettier-plugin-java/test/unit-test/blank_lines/_input.java
@@ -11,9 +11,15 @@ public class BlankLines {
   // Bug Fix: https://github.com/jhipster/prettier-java/issues/368
   private String fieldOne;
   private String fieldTwo;
-  private @Nullable String shouldAddLineBeforeAndAfter;
+  @Nullable private String shouldAddLineBeforeAndAfter;
   private String fieldThree;
   private String fieldFour;
+
+  private String a;
+  private String b;
+  private @Nullable String shouldNotAddBlankLines;
+  private String d;
+  private String e;
 
   public int m = 4;
   public Constructors() {

--- a/packages/prettier-plugin-java/test/unit-test/blank_lines/_output.java
+++ b/packages/prettier-plugin-java/test/unit-test/blank_lines/_output.java
@@ -16,6 +16,12 @@ public class BlankLines {
   private String fieldThree;
   private String fieldFour;
 
+  private String a;
+  private String b;
+  private @Nullable String shouldNotAddBlankLines;
+  private String d;
+  private String e;
+
   public int m = 4;
 
   public Constructors() {

--- a/packages/prettier-plugin-java/test/unit-test/comments/class/_output.java
+++ b/packages/prettier-plugin-java/test/unit-test/comments/class/_output.java
@@ -671,8 +671,7 @@ public final class ArrayTable<R, C, V>
     return columnKeyToIndex.keySet();
   }
 
-  @MonotonicNonNull
-  private transient ColumnMap columnMap;
+  private transient @MonotonicNonNull ColumnMap columnMap;
 
   @Override
   public Map<C, Map<R, V>> columnMap() {
@@ -762,8 +761,7 @@ public final class ArrayTable<R, C, V>
     return rowKeyToIndex.keySet();
   }
 
-  @MonotonicNonNull
-  private transient RowMap rowMap;
+  private transient @MonotonicNonNull RowMap rowMap;
 
   @Override
   public Map<R, Map<C, V>> rowMap() {

--- a/packages/prettier-plugin-java/test/unit-test/modifiers/_input.java
+++ b/packages/prettier-plugin-java/test/unit-test/modifiers/_input.java
@@ -31,6 +31,14 @@
 final @AnnotationOne static public @AnnotationTwo class ClassWithModifiers {
   transient @AnnotationOne final private @AnnotationTwo static String CONSTANT = "abc";
 
+  final @AnnotationOne static @AnnotationTwo protected @AnnotationThree String CONSTANT_2 = "123";
+
+  static @AnnotationOne public @AnnotationTwo String staticField;
+
+  public @AnnotationOne @AnnotationTwo String twoTrailingAnnotations;
+
+  @AnnotationOne String onlyAnnotations;
+
   final @AnnotationOne static @AnnotationTwo synchronized protected @AnnotationThree String method() {
     return CONSTANT;
   }

--- a/packages/prettier-plugin-java/test/unit-test/modifiers/_output.java
+++ b/packages/prettier-plugin-java/test/unit-test/modifiers/_output.java
@@ -49,6 +49,18 @@ public static final class ClassWithModifiers {
 
   @AnnotationOne
   @AnnotationTwo
+  protected static final @AnnotationThree String CONSTANT_2 = "123";
+
+  @AnnotationOne
+  public static @AnnotationTwo String staticField;
+
+  public @AnnotationOne @AnnotationTwo String twoTrailingAnnotations;
+
+  @AnnotationOne
+  String onlyAnnotations;
+
+  @AnnotationOne
+  @AnnotationTwo
   protected static final synchronized @AnnotationThree String method() {
     return CONSTANT;
   }

